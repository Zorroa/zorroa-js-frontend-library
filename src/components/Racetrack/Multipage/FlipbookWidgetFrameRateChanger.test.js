/* eslint-env jest */

import React from 'react'
import {
  FlipbookWidgetFrameRateChanger,
  mapStateToProps,
} from './FlipbookWidgetFrameRateChanger'
import renderer from 'react-test-renderer'

function generateActions() {
  const setFlipbookFps = jest.fn()
  const actions = {
    setFlipbookFps,
  }

  return actions
}

function generateProps(customProps) {
  return {
    setFlipbookFps: 12,
    actions: generateActions(),
    ...customProps,
  }
}

describe('<FlipbookWidgetFrameRateChanger />', () => {
  describe('mapStateToProps', () => {
    it('Should map the state to a new, simpler object', () => {
      expect(
        mapStateToProps({
          app: {
            flipbookFps: 12,
          },
        }),
      ).toEqual({
        fps: 12,
      })
    })
  })

  describe('At 12 FPS', () => {
    it('Should render the toggle bar a third of the way through', () => {
      const props = generateProps()
      const tree = renderer
        .create(<FlipbookWidgetFrameRateChanger {...props} />)
        .toJSON()
      expect(tree).toMatchSnapshot()
    })
  })

  describe('At 24 FPS', () => {
    it('Should render the toggle bar half of the way through', () => {
      const props = generateProps({
        fps: 24,
      })
      const tree = renderer
        .create(<FlipbookWidgetFrameRateChanger {...props} />)
        .toJSON()
      expect(tree).toMatchSnapshot()
    })
  })

  describe('At 30 FPS', () => {
    it('Should render the toggle bar all the way through', () => {
      const props = generateProps({
        fps: 30,
      })
      const tree = renderer
        .create(<FlipbookWidgetFrameRateChanger {...props} />)
        .toJSON()
      expect(tree).toMatchSnapshot()
    })
  })
})
