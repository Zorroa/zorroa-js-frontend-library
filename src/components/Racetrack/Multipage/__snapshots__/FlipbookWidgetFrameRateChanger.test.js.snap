// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<FlipbookWidgetFrameRateChanger /> At 12 FPS Should render the toggle bar a third of the way through 1`] = `
<div
  className="FlipbookWidgetFrameRateChanger__fps-selector FlipbookWidgetFrameRateChanger__section"
>
  <div
    className="FlipbookWidgetFrameRateChanger__fps-title"
  >
    Flipbook frame rate
  </div>
  <div
    className="FlipbookWidgetFrameRateChanger__fps-grabber-container"
  >
    <div
      className="FlipbookWidgetFrameRateChanger__fps-grabber"
      style={
        Object {
          "width": "calc(0 / 3 * 100%)",
        }
      }
    >
      <div
        className="FlipbookWidgetFrameRateChanger__fps-grabber-target"
        style={
          Object {
            "left": "calc(-1 / 3 * 100%)",
            "width": "calc(1 / 3 * 100%)",
          }
        }
      >
        <div
          className="FlipbookWidgetFrameRateChanger__fps-grabber-target-circle"
        />
      </div>
    </div>
  </div>
  <div
    className="FlipbookWidgetFrameRateChanger__fps-rate-list"
  >
    <div
      className="FlipbookWidgetFrameRateChanger__fps-rate"
      onClick={[Function]}
    >
      12
       FPS
    </div>
    <div
      className="FlipbookWidgetFrameRateChanger__fps-rate"
      onClick={[Function]}
    >
      24
       FPS
    </div>
    <div
      className="FlipbookWidgetFrameRateChanger__fps-rate"
      onClick={[Function]}
    >
      30
       FPS
    </div>
  </div>
</div>
`;

exports[`<FlipbookWidgetFrameRateChanger /> At 24 FPS Should render the toggle bar half of the way through 1`] = `
<div
  className="FlipbookWidgetFrameRateChanger__fps-selector FlipbookWidgetFrameRateChanger__section"
>
  <div
    className="FlipbookWidgetFrameRateChanger__fps-title"
  >
    Flipbook frame rate
  </div>
  <div
    className="FlipbookWidgetFrameRateChanger__fps-grabber-container"
  >
    <div
      className="FlipbookWidgetFrameRateChanger__fps-grabber"
      style={
        Object {
          "width": "calc(2 / 3 * 100%)",
        }
      }
    >
      <div
        className="FlipbookWidgetFrameRateChanger__fps-grabber-target"
        style={
          Object {
            "left": "calc(1 / 3 * 100%)",
            "width": "calc(1 / 3 * 100%)",
          }
        }
      >
        <div
          className="FlipbookWidgetFrameRateChanger__fps-grabber-target-circle"
        />
      </div>
    </div>
  </div>
  <div
    className="FlipbookWidgetFrameRateChanger__fps-rate-list"
  >
    <div
      className="FlipbookWidgetFrameRateChanger__fps-rate FlipbookWidgetFrameRateChanger__fps-rate--elapsed"
      onClick={[Function]}
    >
      12
       FPS
    </div>
    <div
      className="FlipbookWidgetFrameRateChanger__fps-rate FlipbookWidgetFrameRateChanger__fps-rate--active"
      onClick={[Function]}
    >
      24
       FPS
    </div>
    <div
      className="FlipbookWidgetFrameRateChanger__fps-rate"
      onClick={[Function]}
    >
      30
       FPS
    </div>
  </div>
</div>
`;

exports[`<FlipbookWidgetFrameRateChanger /> At 30 FPS Should render the toggle bar all the way through 1`] = `
<div
  className="FlipbookWidgetFrameRateChanger__fps-selector FlipbookWidgetFrameRateChanger__section"
>
  <div
    className="FlipbookWidgetFrameRateChanger__fps-title"
  >
    Flipbook frame rate
  </div>
  <div
    className="FlipbookWidgetFrameRateChanger__fps-grabber-container"
  >
    <div
      className="FlipbookWidgetFrameRateChanger__fps-grabber"
      style={
        Object {
          "width": "calc(3 / 3 * 100%)",
        }
      }
    >
      <div
        className="FlipbookWidgetFrameRateChanger__fps-grabber-target"
        style={
          Object {
            "left": "calc(2 / 3 * 100%)",
            "width": "calc(1 / 3 * 100%)",
          }
        }
      >
        <div
          className="FlipbookWidgetFrameRateChanger__fps-grabber-target-circle"
        />
      </div>
    </div>
  </div>
  <div
    className="FlipbookWidgetFrameRateChanger__fps-rate-list"
  >
    <div
      className="FlipbookWidgetFrameRateChanger__fps-rate FlipbookWidgetFrameRateChanger__fps-rate--elapsed"
      onClick={[Function]}
    >
      12
       FPS
    </div>
    <div
      className="FlipbookWidgetFrameRateChanger__fps-rate FlipbookWidgetFrameRateChanger__fps-rate--elapsed"
      onClick={[Function]}
    >
      24
       FPS
    </div>
    <div
      className="FlipbookWidgetFrameRateChanger__fps-rate FlipbookWidgetFrameRateChanger__fps-rate--active"
      onClick={[Function]}
    >
      30
       FPS
    </div>
  </div>
</div>
`;
